File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        hashSet = set()
        for i in range(9):
            for j in range(9):
                if board[i][j] != ".":
                    isRow = f"{board[i][j]} is row {i}"
                    isColumn = f"{board[i][j]} is column {j}"
                    isBlock = f"{board[i][j]} is block {i // 3}-{j // 3}"
                    if isRow in hashSet or isColumn in hashSet or isBlock in hashSet:
                        return False
                    else:
                        hashSet.add(isRow)
                        hashSet.add(isColumn)
                        hashSet.add(isBlock)
        return True

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                if (board[i][j] != '.') {
                    String isRow = board[i][j] + " is row " + i;
                    String isCol = board[i][j] + " is col " + j;
                    String isBlock = board[i][j] + " is block " + i / 3 + ", " + j / 3;
                    if (map.containsKey(isRow) || map.containsKey(isCol) || map.containsKey(isBlock)) {
                        return false;
                    } else {
                        map.put(isRow, 1);
                        map.put(isCol, 1);
                        map.put(isBlock, 1);
                    }
                }
            }
        }
        return true;
    }
}